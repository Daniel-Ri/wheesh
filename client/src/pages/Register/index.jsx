/* eslint-disable jsx-a11y/label-has-associated-control */
import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { Button, IconButton, InputAdornment, OutlinedInput } from '@mui/material';
import BackBtn from '@components/BackBtn';
import { Visibility, VisibilityOff } from '@mui/icons-material';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import './custom.css';

import toast from 'react-hot-toast';
import { registerUser, sendEmailToken } from '@containers/Client/actions';
import classes from './style.module.scss';

const Register = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const [showPassword, setShowPassword] = useState(false);
  const [showPasswordConfirmation, setShowPasswordConfirmation] = useState(false);

  const [inputs, setInputs] = useState({
    username: '',
    password: '',
    passwordConfirmation: '',
    gender: '',
    dateOfBirth: '',
    idCard: '',
    name: '',
    email: '',
    emailToken: '',
  });

  const handleClickShowPassword = () => setShowPassword((show) => !show);

  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  const handleClickShowPasswordConfirmation = () => setShowPasswordConfirmation((show) => !show);

  const handleMouseDownPasswordConfirmation = (event) => {
    event.preventDefault();
  };

  const handleInputChange = (e) => {
    setInputs((prev) => ({ ...prev, [e.target.name]: e.target.value }));
  };

  const validateUsername = () => {
    if (!inputs.username) {
      toast.error('Username cannot be empty');
      return false;
    }

    return true;
  };

  const validatePassword = () => {
    if (!inputs.password) {
      toast.error('Password cannot be empty');
      return false;
    }

    if (inputs.password.length < 6) {
      toast.error('Password must have min 6 characters');
      return false;
    }

    return true;
  };

  const validatePasswordConfirmation = () => {
    if (!inputs.passwordConfirmation) {
      toast.error('You must insert the password confirmation');
      return false;
    }
    if (inputs.passwordConfirmation !== inputs.password) {
      toast.error('Password confirmation must have same value as password');
      return false;
    }

    return true;
  };

  const validateIdCard = () => {
    if (!inputs.idCard) {
      toast.error('You must insert ID Card');
      return false;
    }
    if (inputs.idCard.length !== 16 || !/^\d+$/.test(inputs.idCard)) {
      toast.error('Incorrect ID Card format');
      return false;
    }

    return true;
  };

  const validateEmail = () => {
    if (!inputs.email) {
      toast.error('You must insert email');
      return false;
    }

    if (!/^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(inputs.email)) {
      toast.error('Incorrect E-mail format');
      return false;
    }

    return true;
  };

  const handleFocusOut = (e) => {
    if (e.target.name === 'username') {
      validateUsername();
    } else if (e.target.name === 'password') {
      validatePassword();
    } else if (e.target.name === 'passwordConfirmation') {
      validatePasswordConfirmation();
    } else if (e.target.name === 'idCard') {
      validateIdCard();
    } else if (e.target.name === 'email') {
      validateEmail();
    }
  };

  const validateInputs = () => {
    if (!validateUsername()) return false;

    if (!validatePassword()) return false;

    if (!validatePasswordConfirmation()) return false;

    if (!validateIdCard()) return false;

    if (!validateEmail()) return false;

    return true;
  };

  const handleSuccessRegister = () => {
    toast.success('Success registration');
    navigate('/login');
  };

  const handleErrorRegister = (errorMsg) => {
    toast.error(errorMsg);
  };

  const handleSubmit = () => {
    if (!validateInputs()) return;

    const formattedInputs = { ...inputs };
    formattedInputs.dateOfBirth = new Date(inputs.dateOfBirth).toDateString();
    delete formattedInputs.passwordConfirmation;

    dispatch(registerUser(formattedInputs, handleSuccessRegister, handleErrorRegister));
  };

  const handleSuccessSendEmailToken = () => {
    toast.success('Sent a token via email');
  };

  const handleErrorSendEmailToken = (errorMsg) => {
    toast.error(errorMsg);
    setInputs((prev) => ({ ...prev, email: '' }));
  };

  const handleGenerateEmailToken = () => {
    if (!validateEmail()) return;

    const formattedInputs = { email: inputs.email, action: 'create' };
    dispatch(sendEmailToken(formattedInputs, handleSuccessSendEmailToken, handleErrorSendEmailToken));
  };

  return (
    <main className={classes.main}>
      <div className={classes.container}>
        <header>
          <BackBtn handleClickBack={() => navigate('/login')} />
          <h1>Register</h1>
        </header>

        <form>
          <div className={classes.header}>Personal Information</div>
          <div className={classes.input}>
            <label htmlFor="username">User Name</label>
            <input
              type="text"
              name="username"
              id="username"
              value={inputs.username}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              placeholder="Enter the username"
              autoComplete="off"
            />
          </div>
          <div className={classes.input}>
            <label htmlFor="password">Password</label>
            <OutlinedInput
              id="outlined-adornment-password"
              name="password"
              placeholder="Enter the password"
              value={inputs.password}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              className={classes.inputPassword}
              type={showPassword ? 'text' : 'password'}
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={handleClickShowPassword}
                    onMouseDown={handleMouseDownPassword}
                    edge="end"
                  >
                    {showPassword ? <VisibilityOff /> : <Visibility />}
                  </IconButton>
                </InputAdornment>
              }
              label=""
            />
          </div>
          <div className={classes.input}>
            <label htmlFor="passwordConfirmation">Password Confirmation</label>
            <OutlinedInput
              id="outlined-adornment-password"
              name="passwordConfirmation"
              placeholder="Enter password confirmation"
              value={inputs.passwordConfirmation}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              className={classes.inputPassword}
              type={showPasswordConfirmation ? 'text' : 'password'}
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={handleClickShowPasswordConfirmation}
                    onMouseDown={handleMouseDownPasswordConfirmation}
                    edge="end"
                  >
                    {showPasswordConfirmation ? <VisibilityOff /> : <Visibility />}
                  </IconButton>
                </InputAdornment>
              }
              label=""
            />
          </div>
          <div className={classes.input}>
            <label htmlFor="gender">Gender</label>
            <div className={classes.radios}>
              <div className={classes.radio} onClick={() => setInputs((prev) => ({ ...prev, gender: 'Male' }))}>
                <input
                  type="radio"
                  name="gender"
                  value="Male"
                  checked={inputs.gender === 'Male'}
                  onChange={handleInputChange}
                />
                <label>Male</label>
              </div>
              <div className={classes.radio} onClick={() => setInputs((prev) => ({ ...prev, gender: 'Female' }))}>
                <input
                  type="radio"
                  name="gender"
                  value="Female"
                  checked={inputs.gender === 'Female'}
                  onChange={handleInputChange}
                />
                <label>Female</label>
              </div>
            </div>
          </div>
          <div className={classes.input}>
            <label htmlFor="dateOfBirth">Date of birth</label>
            <DatePicker
              name="dateOfBirth"
              placeholderText="Select your date of birth"
              selected={inputs.dateOfBirth}
              showMonthDropdown
              dropdownMode="select"
              showYearDropdown
              dateFormat="dd/MM/yyyy"
              className={classes.datePicker}
              onChange={(date) => setInputs((prev) => ({ ...prev, dateOfBirth: date }))}
            />
          </div>

          <div className={classes.header}>Certificate Information</div>
          <div className={classes.input}>
            <label htmlFor="idCard">ID Card</label>
            <input
              type="text"
              name="idCard"
              id="idCard"
              value={inputs.idCard}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              placeholder="Please enter your Indonesia ID card"
              autoComplete="off"
            />
          </div>
          <div className={classes.input}>
            <label htmlFor="name">Name</label>
            <input
              type="text"
              name="name"
              id="name"
              value={inputs.name}
              onChange={handleInputChange}
              placeholder="Enter your name on your ID Card"
              autoComplete="off"
            />
          </div>

          <div className={classes.header}>Contact Information</div>
          <div className={classes.input}>
            <label htmlFor="email">E-mail</label>
            <input
              type="text"
              name="email"
              id="email"
              value={inputs.email}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              placeholder="Please enter your email address"
              autoComplete="off"
            />
          </div>
          <div className={classes.input}>
            <input
              className={classes.verification}
              type="text"
              name="emailToken"
              id="emailToken"
              value={inputs.emailToken}
              onChange={handleInputChange}
              placeholder="Please enter the verification code"
              autoComplete="off"
            />
            <div className={classes.button}>
              <Button variant="contained" className={classes.btn} onClick={handleGenerateEmailToken}>
                Generate
              </Button>
            </div>
          </div>
        </form>

        <div className={classes.buttons}>
          <Button variant="contained" className={classes.submit} onClick={handleSubmit}>
            Submit
          </Button>
        </div>
      </div>
    </main>
  );
};

export default Register;
