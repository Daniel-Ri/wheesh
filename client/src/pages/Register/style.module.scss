@import '@styles/base';

// Reset
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: Poppins;
}

.main {
  margin-left: 200px;
  position: relative;
  background: $color-alabaster;
  display: flex;
  justify-content: center;
  width: calc(100vw - 200px);
  height: 100%;
  min-height: 100vh;

  .container {
    width: 90%;
    max-width: 40rem;
    @include directionY;

    header {
      width: 100%;
      margin-top: 5vh;
      @include directionX;
      justify-content: center;
      align-items: center;
      position: relative;

      h1 {
        font-size: 120%;
      }
    }

    form {
      margin-top: 1rem;
      width: 100%;
      @include directionY;

      .header {
        padding: 0.5rem;
        background-color: $color-light-gray;
      }

      label {
        flex: 45%;
      }

      div.input {
        width: 100%;
        padding: 0.5rem;
        @include directionX;
        justify-content: space-between;
        align-items: center;
        gap: 0.5rem;

        input {
          flex: 45%;
          width: 45%;
          padding: 0.25rem 0.5rem;
          font-size: 80%;
          text-align: right;
          border-radius: 5px;
        }
  
        input:focus {
          outline: none;
          border-color: $color-picton-blue;
        }
  
        .inputPassword {
          flex: 45%;
          width: 45%;
          font-size: 100%;
        }
  
        .radios {
          flex: 45%;
          width: 45%;
          @include directionX;
          justify-content: space-around;
          gap: 1rem;
  
          .radio {
            @include directionX;
            gap: 0.25rem;
            cursor: pointer;
  
            input {
              cursor: pointer;
            }
  
            label {
              cursor: pointer;
            }
          }
        }
  
        .datePickerWrapper {
          @include directionX;
          flex: 45%;
  
          .datePicker {
            align-self: flex-end;
          }
        }
  
        .verification {
          flex: 45%;
          width: 45%;
          text-align: left;
        }
  
        .button {
          align-self: flex-end;
          @include directionX;
          justify-content: flex-end;
          flex: 45%;
        }
      }

      .errorRowReverse {
        width: 100%;
        margin-top: -0.5rem;
        padding: 0 0.5rem;
        @include directionX;
        justify-content: flex-end;
        color: $color-strawberry-red;
        font-size: 70%;

        .errorMsg {
          text-align: right;
        }
      }

      .errorRow {
        width: 100%;
        margin-top: -0.5rem;
        padding: 0 0.5rem;
        @include directionX;
        color: $color-strawberry-red;
        font-size: 70%;
      }
    }

    .buttons {
      margin-top: 1rem;
      margin-bottom: 1rem;
      @include directionY;
      gap: 1rem;
    }

    .mainError {
      margin-top: 0.5rem;
      margin-bottom: -0.5rem;
      text-align: center;
      color: $color-strawberry-red;
      font-size: 70%;
    }
  }

  @media screen and (max-width: $screen-lg-limit) {
    margin-left: 192px;
    width: calc(100vw - 192px);
  }

  @media screen and (max-width: $screen-md-limit) {
    margin-left: 10rem;
    width: calc(100vw - 10rem);
  }

  @media screen and (max-width: $screen-sm-limit) {
    margin-left: 0;
    width: 100vw;
    margin-bottom: 76px;
    height: 100%;
    min-height: calc(100vh - 76px);

    .container {
      header {
        height: 3rem;
      }
    }
  }
}