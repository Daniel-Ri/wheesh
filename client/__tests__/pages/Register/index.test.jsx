import { render, fireEvent } from '@utils/testHelper';
import Register from '@pages/Register';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let wrapper;

const mockProps = {
  login: false,
};

beforeEach(() => {
  wrapper = render(<Register {...mockProps} />);
});

describe('Page Register', () => {
  test('Renders Register page', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('Register')).toBeInTheDocument();
  });

  test('Should call navigate when BackBtn clicked', () => {
    const { getByTestId } = wrapper;
    const backButton = getByTestId('BackBtn');
    fireEvent.click(backButton);
    expect(mockNavigate).toHaveBeenCalledWith('/login');
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
