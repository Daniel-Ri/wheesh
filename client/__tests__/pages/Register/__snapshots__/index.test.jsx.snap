// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Register Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
        data-testid="Register"
      >
        <div
          class="container"
        >
          <header>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              data-testid="BackBtn"
              tabindex="0"
              type="button"
            >
              Kembali
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <h1>
              Register
            </h1>
          </header>
          <form>
            <div
              class="header"
            >
              Informasi Pribadi
            </div>
            <div
              class="input"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                autocomplete="off"
                id="username"
                name="username"
                placeholder="Masukkan username"
                type="text"
                value=""
              />
            </div>
            <div
              class="input"
            >
              <label
                for="password"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd inputPassword css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="outlined-adornment-password"
                  name="password"
                  placeholder="Masukkan password"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="input"
            >
              <label
                for="passwordConfirmation"
              >
                Konfirmasi Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd inputPassword css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="outlined-adornment-password"
                  name="passwordConfirmation"
                  placeholder="Masukkan konfirmasi password"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="input"
            >
              <label
                for="gender"
              >
                Jenis Kelamin
              </label>
              <div
                class="radios"
              >
                <div
                  class="radio"
                >
                  <input
                    name="gender"
                    type="radio"
                    value="Male"
                  />
                  <label>
                    Pria
                  </label>
                </div>
                <div
                  class="radio"
                >
                  <input
                    name="gender"
                    type="radio"
                    value="Female"
                  />
                  <label>
                    Wanita
                  </label>
                </div>
              </div>
            </div>
            <div
              class="input"
            >
              <label
                for="dateOfBirth"
              >
                Tanggal Kelahiran
              </label>
              <div
                class="datePickerWrapper"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="datePicker"
                      name="dateOfBirth"
                      placeholder="Pilih tanggal lahir"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="header"
            >
              Informasi Sertifikat
            </div>
            <div
              class="input"
            >
              <label
                for="idCard"
              >
                Kartu Identitas
              </label>
              <input
                autocomplete="off"
                id="idCard"
                name="idCard"
                placeholder="Silakan masukkan KTP Indonesia Anda"
                type="text"
                value=""
              />
            </div>
            <div
              class="input"
            >
              <label
                for="name"
              >
                Nama
              </label>
              <input
                autocomplete="off"
                id="name"
                name="name"
                placeholder="Masukkan nama Anda sesuai KTP"
                type="text"
                value=""
              />
            </div>
            <div
              class="header"
            >
              Informasi Kontak
            </div>
            <div
              class="input"
            >
              <label
                for="email"
              >
                E-mail
              </label>
              <input
                autocomplete="off"
                id="email"
                name="email"
                placeholder="Silakan masukkan alamat email Anda"
                type="text"
                value=""
              />
            </div>
            <div
              class="input"
            >
              <input
                autocomplete="off"
                class="verification"
                id="emailToken"
                name="emailToken"
                placeholder="Silakan masukkan kode verifikasi"
                type="text"
                value=""
              />
              <div
                class="button"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium btn css-sghohy-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Hasilkan
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </form>
          <div
            class="buttons"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium submit css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Kirim
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
      data-testid="Register"
    >
      <div
        class="container"
      >
        <header>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="BackBtn"
            tabindex="0"
            type="button"
          >
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h1>
            Register
          </h1>
        </header>
        <form>
          <div
            class="header"
          >
            Informasi Pribadi
          </div>
          <div
            class="input"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              autocomplete="off"
              id="username"
              name="username"
              placeholder="Masukkan username"
              type="text"
              value=""
            />
          </div>
          <div
            class="input"
          >
            <label
              for="password"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd inputPassword css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-adornment-password"
                name="password"
                placeholder="Masukkan password"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="input"
          >
            <label
              for="passwordConfirmation"
            >
              Konfirmasi Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd inputPassword css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-adornment-password"
                name="passwordConfirmation"
                placeholder="Masukkan konfirmasi password"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="input"
          >
            <label
              for="gender"
            >
              Jenis Kelamin
            </label>
            <div
              class="radios"
            >
              <div
                class="radio"
              >
                <input
                  name="gender"
                  type="radio"
                  value="Male"
                />
                <label>
                  Pria
                </label>
              </div>
              <div
                class="radio"
              >
                <input
                  name="gender"
                  type="radio"
                  value="Female"
                />
                <label>
                  Wanita
                </label>
              </div>
            </div>
          </div>
          <div
            class="input"
          >
            <label
              for="dateOfBirth"
            >
              Tanggal Kelahiran
            </label>
            <div
              class="datePickerWrapper"
            >
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class="datePicker"
                    name="dateOfBirth"
                    placeholder="Pilih tanggal lahir"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="header"
          >
            Informasi Sertifikat
          </div>
          <div
            class="input"
          >
            <label
              for="idCard"
            >
              Kartu Identitas
            </label>
            <input
              autocomplete="off"
              id="idCard"
              name="idCard"
              placeholder="Silakan masukkan KTP Indonesia Anda"
              type="text"
              value=""
            />
          </div>
          <div
            class="input"
          >
            <label
              for="name"
            >
              Nama
            </label>
            <input
              autocomplete="off"
              id="name"
              name="name"
              placeholder="Masukkan nama Anda sesuai KTP"
              type="text"
              value=""
            />
          </div>
          <div
            class="header"
          >
            Informasi Kontak
          </div>
          <div
            class="input"
          >
            <label
              for="email"
            >
              E-mail
            </label>
            <input
              autocomplete="off"
              id="email"
              name="email"
              placeholder="Silakan masukkan alamat email Anda"
              type="text"
              value=""
            />
          </div>
          <div
            class="input"
          >
            <input
              autocomplete="off"
              class="verification"
              id="emailToken"
              name="emailToken"
              placeholder="Silakan masukkan kode verifikasi"
              type="text"
              value=""
            />
            <div
              class="button"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium btn css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Hasilkan
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
        <div
          class="buttons"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium submit css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Kirim
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
